pkg/cmd/grafana/apiserver/server.go

```diff
@@ -101,6 +101,20 @@ func (o *APIServerOptions) Config() (*genericapiserver.RecommendedConfig, error)
 		return nil, fmt.Errorf("failed to apply options to server config: %w", err)
 	}
 
+	if !o.Options.ExtraOptions.DevMode {
+		// When not in DevMode, we don't set up the permission requestheader auth handler
+		// Instead, we wrap it to additionally perform double-authn
+		// TODO: should we allow discovery without needing double-authn, I think we have to
+		validator := auth.NewValidator(o.Options.AuthnOptions.IDVerifierConfig)
+		idTokenAuthenticator := auth.NewIDTokenAuthenticator(validator)
+		serverConfig.Authentication.Authenticator = auth.AppendToAuthenticators(idTokenAuthenticator)
+
+		if serverConfig.Authentication.RequestHeaderConfig != nil {
+			requestHeaderAuthenticator := auth.NewWrappedRequestHeaderAuthenticator(serverConfig, validator)
+			serverConfig.Authentication.Authenticator = auth.AppendToAuthenticators(requestHeaderAuthenticator.AuthenticateRequest, serverConfig.Authentication.Authenticator)
```

Comment:
While making the change for just going with existence of signing URLs instead of DevMode, just don't add this one or it will break aggregation, until I have had a chance to open discovery access and not require an ID token from the parent for that.

Reply:
Which means, keep the original serverConfig.Authentication.Authenticator in the previous AppendTo call.